agent/consul/intention_endpoint.go

```diff
@@ -77,6 +77,10 @@ func (s *Intention) Apply(args *structs.IntentionRequest, reply *string) error {
 		return ErrConnectNotEnabled
 	}
 
+	if args.Intention.SourcePeer != "" {
+		return fmt.Errorf("SourcePeer field is not supported on this endpoint. Use config entries instead")
+	}
```

Comment:
ah yeah, great call out!

@rboyer , why should it live after the `ForwardRPC` call? Wouldn't it be best to fail fast rather than waste another hop?

```
if args.Intention != nil && ...has SourcePeer... {
   // err
}
```

Reply:
Let me know what y'all think here

Reply:
In almost all the other RPC endpoints we forward the request before doing any kind of logic, including validation, so it would mostly be for consistency there. I don't think it's wrong to put it first though - it is quicker/less load on the leader to fail before forwarding the request here.

Reply:
I vaguely remember asking this same question around last year 😆 One side effect of validating before the forward is that the leader will validate again, which is probably negligible unless the validation is costly